script/generate-component-index

```diff
@@ -23,3 +24,22 @@ do
 done
 
 echo "};" >> $target
+
+# Build sass index
+target=./src/styles/components.scss
+
+echo "// Generated by $0" > $target
+echo "@import './colors.scss';" >> $target
+echo "" >> $target
+
+for f in $( find src/components -name '*.scss' | sort )
+do
+    base=$( basename "$f" )
+    echo "@import '../components/$base';" >> $target
+done
+
+for f in $( find src/pages -name '*.scss' | sort )
+do
+    base=$( basename "$f" )
+    echo "@import '../pages/$base';" >> $target
+done
```

Comment:
You can exclude a dir.

On Friday, 21 August 2015, Daniel Heath <notifications@github.com> wrote:

> In script/generate-component-index
> <https://github.com/ask-izzy/ask-izzy/pull/72#discussion_r37605517>:
>
> > +
> > +echo "// Generated by $0" > $target
> > +echo "@import './colors.scss';" >> $target
> > +echo "" >> $target
> > +
> > +for f in $( find src/components -name '*.scss' | sort )
> > +do
> > +    base=$( basename "$f" )
> > +    echo "@import '../components/$base';" >> $target
> > +done
> > +
> > +for f in $( find src/pages -name '*.scss' | sort )
> > +do
> > +    base=$( basename "$f" )
> > +    echo "@import '../pages/$base';" >> $target
> > +done
>
> That would include src/styles, which has (eg) color variables that have
> to be included up-front.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ask-izzy/ask-izzy/pull/72/files#r37605517>.
>